src/link/MachO/Dylib.zig

```diff
@@ -751,8 +751,14 @@ pub const TargetMatcher = struct {
             .v3 => |v3| blk: {
                 var targets = std.ArrayList([]const u8).init(arena.allocator());
                 for (v3.archs) |arch| {
-                    const target = try std.fmt.allocPrint(arena.allocator(), "{s}-{s}", .{ arch, v3.platform });
-                    try targets.append(target);
+                    if (mem.eql(u8, v3.platform, "zippered")) {
```

Comment:
Are we sure this is v3 only thing?

Reply:
and llvm guards it with TBD_V3 here:
```cpp
  if (Scalar == "zippered") {
    if (Ctx && Ctx->FileKind == FileType::TBD_V3) {
      Values.insert(PLATFORM_MACOS);
      Values.insert(PLATFORM_MACCATALYST);
      return {};
    }
    return "invalid platform";
  }
```

note: they also take the same approach as this PR, when reading .tbd v3 to synthesize `zippered` to two v4's target platforms [ `MACOS`, `MACCATALYST` ]

Reply:
Nice, thanks for checking!